### DISCUSSION TITLE
Help with new beta group behavior

### DISCUSSION THREAD

**Discussion by chasut** [music-assistant/support#3047] [Q&A] [answered]
Updated to the latest beta 2.3.0b30 as soon as it came available as I have been awaiting the fix for Apple Music.  Having an issue with that but my biggest concern is the new group behavior.  My setup depends on an airplay group with basically all of the speakers in the house to manage my music through HA.  So Selecting a speaker (or group of speakers) would play to those speakers and all of the meta-data would be available on the group player for display in my HA front-end.

Now it appears that the group player is all or nothing.  If I turn on the group player, ALL of the speakers in that group turn on (which would happen before in certain circumstances), but I cannot turn ANY of the speakers in the group off.  If I turn on one of the individual speakers, the group stays off.  There are two new concepts with Player groups in player providers, as well as something called "dynamic members" in the existing groups.  I just have no idea if it is working as designed or not. 

As Beta, I understand that documentation is forthcoming.  But would appreciate at least a little understanding of how the new group structure is supposed to work.  

Thanks

**Comment by OzGav** [comment]
The docs describe the group options https://music-assistant.io/faq/groups/

Dynamic members of the sync groups was a late addition so I haven’t played with it yet. Hence the large disclaimer against that. 

You can mute any players you wish.

You can use HA automations to join any series of players you wish into Manual Sync groups.

**Comment by chasut** [comment]
Thank you for the reply.  I have been playing around with this a bit more and I am still baffled.  My setup relies on the feature "In addition to playing in sync the group will hold the queue regardless of which player(s) in the group are powered off" which is no longer the case.  I tried just muting, but not all of my speakers support mute, and it is really not how I would like to manage my whole home audio.

What I need is a consistent player/speaker exposed to HA that ALWAYS has the current playing/queue information, irrespective of which speakers are playing that queue.  If I am just doing it wrong, please give me a better way.

As I am also having an issue with Apple Music in the new beta, I think maybe my best move at this point is to scrap everything and re-build from the release branch and pray that the new group behavior evolves to something that works for me.

Please advise.
Thanks for your help

**Reply by OzGav** [reply]
I don’t know why you say that SYNC GROUPS no longer hold the queue regardless of which players are powered off. That is exactly how they work. FYI there is a sync leader elected and if that player is lost then playback will stop but the queue will not be lost and can be restarted.

I believe you have two options if you don’t want to use mute broadly.
1. Use the experimental dynamic members option so you can add and remove players from the group. Consider this feature in ALPHA stage as we have done minimal internal testing.
2. Identify a player that you can mute, use MANUAL SYNC and join and unjoin whichever players you like to that player. If you get to the stage where you don’t want that player playing then mute it.

**Comment by chasut** [comment]
Thank you, this is helpful in getting a better understanding of how things are supposed to work.

 On option 2.  Yes, this was my solution when I first started working with MA.  It works fine, just feels a little clunky in the UI.  Once I discovered the Group Player, it worked so well and seemed the obvious solution, so I went that direction.

On option 1.  To be honest, the description is a little vague, and I haven't experimented enough.  The link to docs gets a 401.  And as an ALPHA, it may not be great for WAF.  

BTW, any foresight on what the Playergroup is going.  The description is intriguing, but doesnt look like it is built out yet.

The SYNC Groups no longer hold the queue regardless of which players are powered off because it is no longer possible to power off individual players in the group.  It is now an all or nothing proposition.  I can force one to power off through the api, but the player stops.  If I restart the player, the one I turned off turns back on.

**Comment by chasut** [comment]
After updating to 2.3.0b31 I am able to use the Sync Group as I did before.  The difference is that instead of just powering on/off each of the individual speakers, I am joining the group using the HA media_player.join action targeting my Sync Group player.  For some reason I remember that the join/unjoin didn't work in the old sync groups but really don't recall.  I think this will work better as it appears to follow the HA media group concept (as far as I know anyway).
 I do have the "enable dynamic members" feature enabled, although I haven't tested enough at this point to understand what that does or doesn't do in my use case.

In any case, still playing and learning.  Sharing in case others are working through similar use case.

**Reply by OzGav** [reply]
The idea of a sync group is for a permanent group of players that are continuously playing music. If a player is no longer desired to be used it can be muted. If there is any need to add or remove players then use a manual sync group or try the experimental dynamic members option. If you use the dynamic members option then at power on/off of the sync group it will revert to its original members.

p.s. the docs are working fine for me

**Reply by marcelveldt** [reply, marked as answer]
What you are looking for is exactly the dynamic members option and i dont agree with @OzGav here that this is in alpha stage. It is exactly the behavior from how it was before (but some people didnt understand that hence its now got its own toggle).

So in your case, enable the dynamic members option on the group and you can unjoin or power ff players as you wish

**Comment by chasut** [comment]
Thanks to both of you.  It is great that it works for me and my home system, but even better to know that it is an intentional functionality.
